The-Chonky-Panda/Atelier
# Error handling

## Description
Very basic React error handling. Added an ErrorBoundary component wrapper in /common. Only wrapped ImageGallery with ErrorBoundary for now because of an app breaking bug when clicking on certain related products. App no longer breaks

## How to test
Manually. Navigate to product 47421 click on the 5th thumbnail in the image gallery and then on Bright Future Sunglasses (47422) in Related Products. This product doesn't have that many photos and throws an error. The image gallery should render an error message and a button to refresh the page without breaking the app.

## Notes
- I could probably be more comprehensive in where I use the error boundary but this works for now.
- It also seems that the error boundary may trigger with non app-breaking errors and would thus disrupt the user experience unnecessarily.
- I have another branch to fix this specific bug but wanted to hold off on a PR so I could use it as an example here

## Issue tracking
https://trello.com/c/rStGqnLP/143-m-add-proper-error-handling-throughout-anything-widget-specific
